maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/extensions/E2ETest.java

```diff
@@ -184,4 +192,74 @@ public void close()
             .isPositive()
             .isLessThanOrEqualTo( 6_000L );
     }
+
+    @Test( timeout = 10_000L )
+    public void shouldAuthenticateClient() throws Exception
+    {
+        ForkNodeArguments forkNodeArguments = mock( ForkNodeArguments.class );
+        when( forkNodeArguments.getSessionId() ).thenReturn( UUID.randomUUID() );
+
+        try ( SurefireForkChannel server = new SurefireForkChannel( forkNodeArguments ) )
+        {
+            Thread t = new Thread()
+            {
+                @Override
+                public void run()
+                {
+                    SurefireMasterProcessChannelProcessorFactory client
+                        = new SurefireMasterProcessChannelProcessorFactory();
+                    try
+                    {
+                        client.connect( server.getForkNodeConnectionString() );
+                    }
+                    catch ( IOException e )
+                    {
+                        e.printStackTrace();
```

Comment:
Wouldn't this be handled anyway when the runtime exception is catched by JUnit?

Reply:
The view in GH is not enough. You have to see whole nested class `Runnable`.

Reply:
See the [commit](https://github.com/apache/maven-surefire/pull/300/commits/c6f52fbb756482b0dfc3e5fae73c434b192db1fd) which explains that we did not expect an exception within the Thread. We expect it in server's facility verifying the session id returned by the client. Now the client's Thread also has to be running without exception.

Reply:
Sorry, I am blind to see the obvious. Can you please link to the line?

Reply:
@michael-o see whole block of code
https://github.com/apache/maven-surefire/blob/c6f52fbb756482b0dfc3e5fae73c434b192db1fd/maven-surefire-common/src/test/java/org/apache/maven/plugin/surefire/extensions/E2ETest.java#L235-L259